src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotRemoveBaseTypeOrInterfaceTests.cs

```diff
@@ -0,0 +1,310 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class CannotRemoveBaseTypeOrInterfaceTests
+    {
+        [Fact]
+        public void PromotedBaseClassOrInterfaceIsNotReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First : FirstBase, IFirstInterface { }
+  public class Second : SecondBase { }
+  public class SecondBase { }
+  public class FirstBase { }
+  public interface IFirstInterface { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First : NewBase { }
+  public class Second : NewSecondBase { }
+  public class NewBase : FirstBase, INewInterface { }
```

Comment:
Can we also test pushing a method down while inserting a base class in the heirarchy?

Reply:
You would expect that to not be flagged right?

Reply:
Right, it's releated to this.  Often times the reason for introducing a base class in the heirarchy is so that you can share some (but not all) of the members of a type.  In such a case you create a new base type, and move some of the members down from your original type.  It's a binary compatible change but one I expect we need to handle in the member comparison as well.

Reply:
I'll make sure that works then. Thanks for the suggestion!

Reply:
Added test and found a bug on `MembersMustExist` I was doing a bad assumption using left type to search on the base types' members when I should've right containing type base types.